frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx

```diff
@@ -0,0 +1,196 @@
+
+interface VolumeClaimTemplateFormProps {
+  name: string;
+  initialSizeValue?: string;
+  initialSizeUnit?: string;
+  initialVolumeMode?: string;
+}
+
+interface RequestSize {
+  value: string;
+  unit: string;
+}
+
+const VolumeClaimTemplateForm: React.FC<VolumeClaimTemplateFormProps> = ({
+  name,
+  initialSizeValue = '1',
+  initialSizeUnit = 'Gi',
+  initialVolumeMode = 'Filesystem',
+}) => {
+  const { t } = useTranslation();
+  const [field] = useField(name);
+  const initAccessModeHelp = t('pipelines-plugin~Permissions to the mounted drive.');
+  const [accessModeHelp, setAccessModeHelp] = React.useState(initAccessModeHelp);
+  const { setFieldValue, setFieldTouched, errors } = useFormikContext<FormikValues>();
+  const [allowedAccessModes, setAllowedAccessModes] = React.useState(initialAccessModes);
+  const [volumeMode, setVolumeMode] = React.useState(initialVolumeMode);
+  const [accessMode, setAccessMode] = React.useState('ReadWriteOnce');
+  const [requestSizeValue, setRequestSizeValue] = React.useState(initialSizeValue);
+  const [requestSizeUnit, setRequestSizeUnit] = React.useState(initialSizeUnit);
+  const [storageProvisioner, setStorageProvisioner] = React.useState('');
+  const [storageClass, setStorageClass] = React.useState(initialSizeUnit);
+  useFormikValidationFix(field.value);
+
+  const handleAccessMode: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    setAccessMode(event.currentTarget.value);
+  };
+
+  const handleVolumeMode: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    setVolumeMode(event.currentTarget.value);
+  };
+
+  const handleStorageClass = (updatedStorageClass): void => {
+    const provisioner: string = updatedStorageClass?.provisioner || '';
+    // if the provisioner is unknown or no storage class selected, user should be able to set any access mode
+    const modes = provisionerAccessModeMapping[provisioner]
+      ? provisionerAccessModeMapping[provisioner]
+      : getAccessModeForProvisioner(provisioner);
+    // setting message to display for various modes when a storage class of a know provisioner is selected
+    const displayMessage =
+      provisionerAccessModeMapping[provisioner] || isCephProvisioner(provisioner)
+        ? t('pipelines-plugin~Access mode is set by storage class and cannot be changed')
+        : t('pipelines-plugin~Permissions to the mounted drive');
+    setAccessMode('ReadWriteOnce');
+    setAccessModeHelp(displayMessage);
+    // setting accessMode to default with the change to Storage Class selection
+    setAllowedAccessModes(modes);
+    setStorageClass(updatedStorageClass?.metadata?.name);
+    setStorageProvisioner(provisioner);
+    if (storageProvisioner.includes(cephStorageProvisioners[1])) {
+      setVolumeMode('Filesystem');
+    }
+  };
+
+  const handleRequestSizeChange = (size: RequestSize): void => {
+    const { value, unit } = size;
+    setRequestSizeValue(value);
```

Comment:
Recommend you just handle the invalid values part of this PR as there is little reason to push it to a future issue.

Suggested change diff:
```
diff --git frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx
index 8f0b6f41e7..b359eb9eb7 100644
--- frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx
+++ frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx
@@ -42,6 +42,7 @@ const VolumeClaimTemplateForm: React.FC<VolumeClaimTemplateFormProps> = ({
   const [allowedAccessModes, setAllowedAccessModes] = React.useState(initialAccessModes);
   const [volumeMode, setVolumeMode] = React.useState(initialVolumeMode);
   const [accessMode, setAccessMode] = React.useState('ReadWriteOnce');
+  const [requestSizeError, setRequestSizeError] = React.useState<string>(null);
   const [requestSizeValue, setRequestSizeValue] = React.useState(initialSizeValue);
   const [requestSizeUnit, setRequestSizeUnit] = React.useState(initialSizeUnit);
   const [storageProvisioner, setStorageProvisioner] = React.useState('');
@@ -80,6 +81,14 @@ const VolumeClaimTemplateForm: React.FC<VolumeClaimTemplateFormProps> = ({

   const handleRequestSizeChange = (size: RequestSize): void => {
     const { value, unit } = size;
+
+    const numberValue = parseInt(value, 10);
+    if (Number.isNaN(numberValue) || numberValue <= 0) {
+      setRequestSizeError(t('pipelines-plugin~Size must be an integer greater than 0.'));
+      setRequestSizeValue('');
+      return;
+    }
+
     setRequestSizeValue(value);
     setRequestSizeUnit(unit);
   };
@@ -171,8 +180,13 @@ const VolumeClaimTemplateForm: React.FC<VolumeClaimTemplateFormProps> = ({
           describedBy="request-size-help"
           inputID="request-size-input"
           data-test-id="pvc-size-input"
-        />
-        <p className="help-block">{t('pipelines-plugin~Desired storage capacity')}</p>
+        >
+          {requestSizeError ? (
+            <p className="pf-c-form__helper-text pf-m-error">{requestSizeError}</p>
+          ) : (
+            <p className="help-block">{t('pipelines-plugin~Desired storage capacity')}</p>
+          )}
+        </RequestSizeInput>
         <label className="control-label" htmlFor="volume-mode">
           {t('pipelines-plugin~Volume Mode')}
         </label>
```

Reply:
This will not disable the form unless the form field in set to an invalid value, I am doing that in line 98